libraries/Adapters/Microsoft.Bot.Builder.Adapters.Webex/WebexAdapter.cs

```diff
@@ -0,0 +1,366 @@
+
+            var json = JsonConvert.SerializeObject(payload);
+
+            if (!string.Equals(this.config.Secret, string.Empty))
+            {
+                var signature = request.Headers["x-spark-signature"];
+
+                using (var hmac = new HMACSHA1(Encoding.UTF8.GetBytes(this.config.Secret)))
+                {
+                    var hashArray = hmac.ComputeHash(Encoding.UTF8.GetBytes(json));
+
+                    string hash = BitConverter.ToString(hashArray).Replace("-", string.Empty).ToLower();
+
+                    if (!string.Equals(signature, hash))
+                    {
+                        throw new Exception("WARNING: Webhook received message with invalid signature. Potential malicious behavior!");
+                    }
+                }
+            }
+
+            Activity activity;
+            if (payload.resource == "messages" && payload["event"] == "created")
+            {
+                Message decryptedMessage = (await this.api.GetMessageAsync(payload.data.id.ToString())).GetData();
+                activity = new Activity()
+                {
+                    Id = decryptedMessage.Id,
+                    Timestamp = new DateTime(),
+                    ChannelId = "webex",
+                    Conversation = new ConversationAccount()
+                    {
+                        Id = (decryptedMessage as dynamic).SpaceId,
+                    },
+                    From = new ChannelAccount()
+                    {
+                        Id = decryptedMessage.PersonId,
+                        Name = decryptedMessage.PersonEmail,
+                    },
+                    Recipient = new ChannelAccount()
+                    {
+                        Id = this.Identity.Id,
+                    },
+                    Text = decryptedMessage.Text,
+                    ChannelData = decryptedMessage,
+                    Type = ActivityTypes.Message,
+                };
+
+                // this is the bot speaking
+                if (activity.From.Id == this.Identity.Id)
+                {
+                    (activity.ChannelData as dynamic).botkitEventType = "self_message";
+                    activity.Type = ActivityTypes.Event;
+                }
+
+                if (decryptedMessage.HasHtml)
+                {
+                    var pattern = new Regex("^(<p>)?<spark-mention .*?data-object-id=" + this.Identity.Id + ".*?>.*?</spark-mention>");
+                    if (!decryptedMessage.Html.Equals(pattern))
+                    {
+                        var encodedId = this.Identity.Id;
+
+                        // this should look like ciscospark://us/PEOPLE/<id string>
+                        Match match = Regex.Match(encodedId, "/ciscospark://.*/(.*)/im");
+                        pattern = new Regex("^(<p>)?<spark-mention .*?data-object-id=" + match.Captures[1] + ".*?>.*?</spark-mention>");
```

Comment:
same. This one should be:
`"^(<p>)?<spark-mention .*?data-object-id=\"" + match.Captures[1] + "\".*?>.*?</spark-mention>"`